### PR TITLE
8310921: Fix -Wconversion warnings from GenerateOopMap

### PR DESCRIPTION
**Pull request by coleenp**
This is broken out from but tested with the PR from https://github.com/openjdk/jdk/pull/14659

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8310921](https://bugs.openjdk.org/browse/JDK-8310921): Fix -Wconversion warnings from GenerateOopMap (**Enhancement** - P4)

### Reviewers
 * [Ioi Lam](https://openjdk.org/census#iklam) (@iklam - **Reviewer**) ⚠️ Review applies to [3902331a](https://git.openjdk.org/jdk/pull/14666/files/3902331a342f6e0c8637adaf38a48776e7f7690a)
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/14666/head:pull/14666` \
`$ git checkout pull/14666`

Update a local copy of the PR: \
`$ git checkout pull/14666` \
`$ git pull https://git.openjdk.org/jdk.git pull/14666/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 14666`

View PR using the GUI difftool: \
`$ git pr show -t 14666`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/14666.diff">https://git.openjdk.org/jdk/pull/14666.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/14666#issuecomment-1608218763)

### CODE REVIEW

**Review by iklam — APPROVED** [review]
LGTM.

File: src/hotspot/share/oops/generateOopMap.cpp [hunk 1]
```diff
@@ -660,7 +660,7 @@ void GenerateOopMap::restore_state(BasicBlock *bb)
 }
 
 int GenerateOopMap::next_bb_start_pc(BasicBlock *bb) {
- int bbNum = bb - _basic_blocks + 1;
+ int bbNum = (int)(bb - _basic_blocks + 1);
```
Comment (dholmes-ora) [inline src/hotspot/share/oops/generateOopMap.cpp:663]: If this is an offset into a memory region then shouldn't it be `intptr_t` or `size_t`?

File: src/hotspot/share/oops/generateOopMap.cpp [hunk 2]
```diff
@@ -2057,7 +2057,7 @@ void GenerateOopMap::print_time() {
   tty->print_cr ("---------------------------");
   tty->print_cr ("  Total : %3.3f sec.", GenerateOopMap::_total_oopmap_time.seconds());
   tty->print_cr ("  (%3.0f bytecodes per sec) ",
-  GenerateOopMap::_total_byte_count / GenerateOopMap::_total_oopmap_time.seconds());
+  (float)GenerateOopMap::_total_byte_count / GenerateOopMap::_total_oopmap_time.seconds());
```
Comment (dholmes-ora) [inline src/hotspot/share/oops/generateOopMap.cpp:2060]: I don't understand this change. The denominator is a double due to `seconds()` so the numerator will be promoted to double as well, and the whole expression is a double.

File: src/hotspot/share/oops/generateOopMap.cpp (hunk 1 above)
Comment (coleenp) [reply src/hotspot/share/oops/generateOopMap.cpp]: I made bbNum an intptr_t.

File: src/hotspot/share/oops/generateOopMap.cpp [hunk 3]
```diff
@@ -2057,7 +2057,7 @@ void GenerateOopMap::print_time() {
   tty->print_cr ("---------------------------");
   tty->print_cr ("  Total : %3.3f sec.", GenerateOopMap::_total_oopmap_time.seconds());
   tty->print_cr ("  (%3.0f bytecodes per sec) ",
-  (float)GenerateOopMap::_total_byte_count / GenerateOopMap::_total_oopmap_time.seconds());
+  (double)GenerateOopMap::_total_byte_count / GenerateOopMap::_total_oopmap_time.seconds());
```
Comment (dholmes-ora) [inline src/hotspot/share/oops/generateOopMap.cpp:2060]: I still don't see a need for the double cast here - the normal numeric promotion rules will make the expression a double. What error is reported for this code?

Comment (coleenp) [reply src/hotspot/share/oops/generateOopMap.cpp:2060]: /hotspot/share/oops/generateOopMap.cpp: In static member function 'static void GenerateOopMap::print_time()':
/hotspot/share/oops/generateOopMap.cpp:2060:19: warning: conversion from 'uint64_t' {aka 'long unsigned int'} to 'double' may change value [-W
conversion]
 2060 |   GenerateOopMap::_total_byte_count / GenerateOopMap::_total_oopmap_time.seconds());
      |   ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~

Comment (dholmes-ora) [reply src/hotspot/share/oops/generateOopMap.cpp:2060]: Ah now I see. Thanks.

**Review by dholmes-ora — APPROVED** [review]
Looks good. Thanks

**Comment by coleenp** [comment]
Thanks Ioi and David for reviewing.
/integrate

**Pull request closed without merging**